manifests/client.pp

```diff
@@ -77,9 +77,12 @@
       ensure => $package_ensure,
       name   => $package_name,
     }
-    $req = Package['snmp-client']
-  } else {
-    $req = undef
+  }
+
+  $req = $::osfamily ? {
+    'RedHat' => [Package['snmp-client'], Package[$snmp::params::package_name]],
```

Comment:
@razorsedge  Looks like you are correct.  I've made a couple changes and submitted a new PR which should fix these issues.  In the end it may be simpler to just create a file resource that ensures /etc/snmp exists.  Installing the net-snmp package on systems that don't really need it is a bit of a waste.